PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-azure-sdk/src/test/java/com/microsoft/azure/toolkit/intellij/azure/sdk/buildtool/ServiceBusReceiveModeCheckTest.java

```diff
@@ -0,0 +1,243 @@
+
+        verifyProblemRegistered(clientName, methodFoundOne, methodFoundTwo, numOfInvocations, prefetchCountValue);
+    }
+
+    /**
+     * This tests the visitor without a receiveMode and with a prefetchCount of 100.
+     * The test verifies that a problem is not registered with the ProblemsHolder.
+     */
+    @Test
+    public void testWithPeekLockAndNoPrefetchCount() {
+
+        String clientName = "ServiceBusReceiverClient";
+        String methodFoundOne = "receiveMode";
+        String methodFoundTwo = "noprefetchCount";
+        int numOfInvocations = 0;
+        String prefetchCountValue = "100";
+
+        verifyProblemRegistered(clientName, methodFoundOne, methodFoundTwo, numOfInvocations, prefetchCountValue);
+    }
+
+    /**
+     * This tests the visitor without a receiveMode and without a prefetchCount.
+     * The test verifies that a problem is not registered with the ProblemsHolder.
+     */
+    @Test
+    public void testWithoutPeekLockAndNoPrefetchCount() {
+
+        String clientName = "servicebus";
+        String methodFoundOne = "notreceiveMode";
+        String methodFoundTwo = "noprefetchCount";
+        int numOfInvocations = 0;
+        String prefetchCountValue = "100";
+
+        verifyProblemRegistered(clientName, methodFoundOne, methodFoundTwo, numOfInvocations, prefetchCountValue);
+    }
+
+    /**
+     * This tests the visitor with a different Azure service.
+     * The test verifies that a problem is not registered with the ProblemsHolder.
+     */
+    @Test
+    public void testDifferentAzureService() {
+
+        String clientName = "notservicebus";
+        String methodFoundOne = "notreceiveMode";
+        String methodFoundTwo = "noprefetchCount";
+        int numOfInvocations = 0;
+        String prefetchCountValue = "100";
+
+        verifyProblemRegistered(clientName, methodFoundOne, methodFoundTwo, numOfInvocations, prefetchCountValue);
+    }
+
+
+    /**
+     * Create a visitor by calling the buildVisitor method of the ServiceBusReceiveModeCheck class.
+     *
+     * @return The visitor created
+     */
+    private JavaElementVisitor createVisitor() {
+        ServiceBusReceiveModeVisitor mockVisitor = new ServiceBusReceiveModeVisitor(mockHolder, true);
+        return mockVisitor;
+    }
+
+
+    /**
+     * This method verifies that a problem is registered with the ProblemsHolder.
+     *
+     * @param clientName         The name of the client eg. ServiceBusReceiverClient
+     * @param methodFoundOne     The first method found, receiveMode
+     * @param methodFoundTwo     The second method found, prefetchCount
+     * @param numOfInvocations   The number of times the registerProblem method is called
+     * @param prefetchCountValue The value of the prefetchCount
+     */
+    private void verifyProblemRegistered(String clientName, String methodFoundOne, String methodFoundTwo, int numOfInvocations, String prefetchCountValue) {
+
+        String azurePackageName = "com.azure";
```

Comment:
I explored reusing mocks for a while to minimize duplicates, including using .reset(), but some reused mocks were getting overwritten, and others did not change their assigned values when they were returned from different methods so I opted for the duplicates -- this is for the recursion section